From d9d4b1b002e6576eedb73a407a7d2a36b59ff4f9 Mon Sep 17 00:00:00 2001
From: Yan Markman <ymarkman@marvell.com>
Date: Wed, 28 Nov 2018 21:11:15 +0200
Subject: [PATCH 0817/1051] Revert "net: mvpp2: do not recycle buf with
 consumed headroom"

This reverts commit e8e602ab3ecb5d9a0a1aa6e79aafd1d755534f0a.
  "net: mvpp2: do not recycle buf with consumed headroom"
It is WA and replaced by real fix.

Change-Id: I355eff05077291f8da3d22e0a4ee74eb159a9450
Signed-off-by: Yan Markman <ymarkman@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/61420
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Stefan Chulski <stefanc@marvell.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c b/drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c
index 67a8e8bcc2ed..b265d89740a8 100644
--- a/drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c
+++ b/drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c
@@ -3266,9 +3266,6 @@ static int mvpp2_recycle_get_bm_id(struct sk_buff *skb)
 	/* Use skb->cloned but not skb_cloned(), skb_header_cloned() */
 	if (skb_shared(skb) || skb->cloned)
 		return -1;
-	/* WA: don't put to recycle the buffer with fully consumed headroom */
-	if (skb_headroom(skb) <= MVPP2_MH_SIZE)
-		return -1;
 	/* Get bm-pool-id */
 	hash &= MVPP2_RXTX_HASH_BMID_MASK;
 	if (hash >= MVPP2_BM_POOLS_NUM)
-- 
2.17.1

